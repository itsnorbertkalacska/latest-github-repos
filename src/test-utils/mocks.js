export const rawRepo = {
  id: 1,
  name: "My Repo",
  description: "This is the description",
  stargazersCount: 111,
  htmlUrl: "github.com/my-repo",
};

const rawRepoB = {
  id: 2,
  name: "Second Repo",
  description: "Lorem Ipsum",
  stargazersCount: 2450,
  htmlUrl: "github.com/this-is-second",
};

export const repo = {
  ...rawRepo,
  isFavourited: false,
};

const repoB = {
  ...rawRepoB,
  isFavourited: true,
};

export const rawRepos = [rawRepo, rawRepoB];

export const repos = [repo, repoB];
