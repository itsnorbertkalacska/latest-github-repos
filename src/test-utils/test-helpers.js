import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";

import { theme } from "../theme";

export const renderWithProviders = (ui) => {
  function Wrapper({ children }) {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
  return render(ui, { wrapper: Wrapper });
};
