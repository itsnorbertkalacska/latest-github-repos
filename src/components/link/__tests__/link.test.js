import { renderWithProviders } from "../../../test-utils";
import { Link } from "../index";

describe("Link", () => {
  it("passes props properly", () => {
    const { container } = renderWithProviders(
      <Link href="/about">My Link</Link>
    );
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("when external link", () => {
    const isExternal = true;

    it("passes props properly", () => {
      const { container } = renderWithProviders(
        <Link href="https://veed.io" isExternal={isExternal}>
          My Link
        </Link>
      );
      // eslint-disable-next-line testing-library/no-node-access
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
