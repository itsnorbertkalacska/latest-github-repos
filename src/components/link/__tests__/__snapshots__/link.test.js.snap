// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link passes props properly 1`] = `
<a
  class="sc-bczRLJ eMETko"
  href="/about"
>
  My Link
</a>
`;

exports[`Link when external link passes props properly 1`] = `
<a
  class="sc-bczRLJ eMETko"
  href="https://veed.io"
  rel="noreferrer"
  target="_blank"
>
  My Link
</a>
`;
